src/ripple/app/misc/impl/AmendmentTable.cpp

```diff
@@ -539,12 +554,36 @@ AmendmentTableImpl::needValidatedLedger (LedgerIndex ledgerSeq)
 }
 
 void
-AmendmentTableImpl::doValidatedLedger (
+AmendmentTableImpl::doValidatedLedger(
     LedgerIndex ledgerSeq,
-    std::set<uint256> const& enabled)
+    std::set<uint256> const& enabled,
+    majorityAmendments_t const& majority)
 {
     for (auto& e : enabled)
         enable(e);
+
+    std::lock_guard sl(mutex_);
+    // Since we have the whole list in `majority`, reset the time flag, even if
+    // it's currently set. If it's not set when the loop is done, then any
+    // prior unknown amendments have lost majority.
+    firstUnsupportedExpected_.reset();
+    for (auto& m : majority)
+    {
+        auto s = add(m.first);
+
+        if (s->enabled)
+            break;
```

Comment:
Don't we want to check if other amendments may have reached and are unsupported. I.e. Shouldn't this be `continue`?

Reply:
Yes. I totally derped here. Fixed.